### ISSUE TITLE
Authentication method caching sha2 password not supported.

### ISSUE THREAD

**Issue by dazzlersa** [pmachapman/mysqltuner#21]
![image](https://user-images.githubusercontent.com/892876/82533358-65210b80-9b43-11ea-8181-360dcf32be44.png)

Hi, what could I possibly do to resolve this issue?

**Comment by pmachapman** [comment]
MySQL Tuner is designed for MySQL 5.0, and this version is not compatible with MySQL 8.0 (some recommendations will be incorrect or inapplicable).

I am working on an update to support current versions of MySQL and MariaDB, but this will take me some time.

In the meantime, if you still wish to use MySQL Tuner with MySQL 8.0, please update your my.ini file to include the following in the `[mysqld]` section:

```ini
[mysqld]
# Use legacy authentication method
default_authentication_plugin=mysql_native_password
```

This may affect some of your other applications' access to the database server, so I do not recommend performing this on a production system without full testing.

**Comment by bjoernv** [comment]
> In the meantime, if you still wish to use MySQL Tuner with MySQL 8.0, please update your my.ini file to include the following in the `[mysqld]` section:
> 
> ```ini
> [mysqld]
> # Use legacy authentication method
> default_authentication_plugin=mysql_native_password
> ```

I tried this and restarted MySQL 8.0. Even with a newly created second admin user I still get the error message "Authentication method 'caching_sha2_password' not supported by any of the available plugins".

**Comment by pmachapman** [comment]
> I tried this and restarted MySQL 8.0. Even with a newly created second admin user I still get the error message "Authentication method 'caching_sha2_password' not supported by any of the available plugins".

Hi @bjoernv, what version of MySQL 8.0 are you running and on what OS?

I would also suggest double checking that you updated the correct my.ini file used by the mysqld process (I have made that mistake once or twice myself when using non-distro builds).

**Comment by bjoernv** [comment]
@pmachapman, I checked again.
> Hi @bjoernv, what version of MySQL 8.0 are you running and on what OS?
> 
> I would also suggest double checking that you updated the correct my.ini file used by the mysqld process (I have made that mistake once or twice myself when using non-distro builds).

I use MySQL 8.0.26 on Windows 10 Pro 21H1

I found the correct my.ini file in "Services":
`"C:\Program Files\MySQL\MySQL Server 8.0\bin\mysqld.exe" --defaults-file="C:\MySQL-Data\my.ini" MySQL80`

"CREATE USER" still creates caching_sha2_password passwords with

`CREATE USER root2 IDENTIFIED BY 'mypassword';`

So I used
```
CREATE USER root2 IDENTIFIED WITH mysql_native_password BY 'mypassword';
GRANT ALL PRIVILEGES ON *.* TO root2;
FLUSH PRIVILEGES;
SHOW CREATE USER root2;
[...]
| CREATE USER 'root2'@'%' IDENTIFIED WITH 'mysql_native_password' AS '*...' REQUIRE NONE PASSWORD EXPIRE DEFAULT ACCOUNT UNLOCK PASSWORD HISTORY DEFAULT PASSWORD REUSE INTERVAL DEFAULT PASSWORD REQUIRE CURRENT DEFAULT |
```

Anyway, the error "Authentication method 'caching_sha2_password' not supported by any of the available plugins" is still there.

**Comment by bjoernv** [comment]
Personally, the work-around would not help me much. I want to use MySQLTuner in a Windows-only production environment and I want to have the extra security of the 'caching_sha2_password' plugin. If I configure 'mysql_native_password' for a short period of time (e.g. 48 hours), I can not guarantee, that no user changes his password in the time frame.

If I understood the issue correctly, a MySQL Connector .NET update helps to resolve the issue.

**Comment by pmachapman** [comment]
> If I understood the issue correctly, a MySQL Connector .NET update helps to resolve the issue.

Yes, that is correct. I have not done this yet however, as this port of MySQLTuner does not officially support MySQL versions greater than 5.0.

I have not updated this port to match the Perl version in many years, so be careful with the recommendations.

**Comment by pmachapman** [comment]
Fixed in commit [58bfac6](https://github.com/pmachapman/mysqltuner/commit/58bfac67deac97e826ebf3e3aa16867b6b2fda8a).

A fix for this issue will be released with 0.8.6.
